.github/codeql/queries/unique-command-use.ql

```diff
@@ -72,6 +72,20 @@
  
    override string getCommandName() { result = this.getArgument(0).(StringLiteral).getValue() }
  }
+
+ /**
+  * A usage of a command from the typescript code, by calling `CommandManager.execute`.
+  */
+ class CommandUsageCommandManagerMethodCallExpr extends CommandUsage, MethodCallExpr {
+   CommandUsageCommandManagerMethodCallExpr() {
+     this.getCalleeName() = "execute" and
+     this.getReceiver().getType().unfold().(TypeReference).getTypeName().getName() = "CommandManager" and
```

Comment:
```suggestion
     this.getReceiver().getType().(TypeAliasReference).getAliasedType().(TypeReference).getTypeName().getName() = "CommandManager" and
```

I think this achieves the same effect. It's a bit more explicit about what kind of types we expect. Though you could argue we want to make it more robust to changes in the typescript code. I don't have any strong opinions about which direction we go.

Reply:
I do think we want to ensure that if we make a type like `type CommandManagerWithFoo = CommandManager & { foo: string }`, we'd still want to detect it. I think that's what `unfold` allows us to do, but `(TypeAliasReference)` would lose that information.